### DISCUSSION TITLE
Extending the DataFetcher code generator

### DISCUSSION THREAD

**Discussion by skorgs** [Netflix/dgs-framework#925] [General]
Hello,
First of all thank you for this awesome project!
We are trying to use code generation to reduce manual code for our developers. We use the generated types and data fetchers as the start of template, and then manually update those fetchers. Our implementation follow a model where all our fetchers call the same internal service. And most of our manual changes are basically copy paste from one fetcher to another. We were hoping to autogenerate this as well. 
We looked at [DatafetcherGenerator.kt](https://github.com/Netflix/dgs-codegen/blob/master/graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DatafetcherGenerator.kt), and were thinking of extending this in our project. It seems it is final. Our project is Java based (and I am not a Kotlin expert). Is there any way to extend this class to support our usecase? I looked at Kotlin extensions, but I have to define those methods as static in Java ? Are there other ways to extend? If not, is it possible to provide extension points (kotlin based or even gradle based)? I would appreciate any ideas here. 
Thanks

**Comment by srinivasankavitha** [comment]
So you simply want to make this class extensible? You plan to fork the project and add your own specialized handling?

**Reply by skorgs** [reply]
I dont mind forking the project, but was hoping the class to be extensible in main. That way we don't have to worry about future releases to the main project.

**Reply by srinivasankavitha** [reply]
Makes sense,  we can make that change.

**Comment by paulbakker** [comment]
Just making it non-final probably isn't enough, because the generator will
not know about any custom implementation. So if we want to make this
extensible (which would be fine), it requires a bit more work.
